example/cmesh/t8_cmesh_hybrid_new.cxx

```diff
@@ -0,0 +1,306 @@
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <sc_options.h>
+#include <sc_flops.h>
+#include <sc_refcount.h>
+#include <t8_schemes/t8_default/t8_default_cxx.hxx>
+#include <t8_schemes/t8_default/t8_default_tet/t8_dtet.h>
+#include <t8_schemes/t8_default/t8_default_prism/t8_dprism.h>
+#include <t8_schemes/t8_default/t8_default_pyramid/t8_dpyramid.h>
+#include <t8_forest/t8_forest.h>
+#include <t8_cmesh/t8_cmesh_examples.h>
+#include <t8_cmesh/t8_cmesh_partition.h>
+#include <t8_cmesh_readmshfile.h>
+#include <t8_cmesh_vtk_writer.h>
+#include <sc_statistics.h>
+
+static int
+t8_basic_hybrid_refine (t8_forest_t forest, t8_forest_t forest_from, t8_locidx_t which_tree, t8_locidx_t lelement_id,
+                        t8_eclass_scheme_c *ts, int is_family, int num_elements, t8_element_t *elements[])
+{
+  int level, id;
+  level = ts->t8_element_level (elements[0]);
+  if (level >= *(int *) t8_forest_get_user_data (forest)) {
+    return 0;
+  }
+  else {
+    switch (ts->t8_element_shape (elements[0])) {
+    case T8_ECLASS_HEX:
+      id = ts->t8_element_child_id (elements[0]);
+      return id % 2 == 0 ? 1 : 0;
+    case T8_ECLASS_TET:
+      id = ts->t8_element_child_id (elements[0]);
+      return id % 2 == 0 ? 1 : 0;
+    case T8_ECLASS_PRISM:
+      id = ts->t8_element_child_id (elements[0]);
+      return id % 2 == 0 ? 1 : 0;
+    case T8_ECLASS_PYRAMID:
+      return 1;
+    default:
+      return 1;
+    }
+  }
+}
+
+static int
+t8_basic_cake_refine (t8_forest_t forest, t8_forest_t forest_from, t8_locidx_t which_tree, t8_locidx_t lelement_id,
+                      t8_eclass_scheme_c *ts, int is_family, int num_elements, t8_element_t *elements[])
+{
+  int level, type;
+  level = ts->t8_element_level (elements[0]);
+  if (level >= *(int *) t8_forest_get_user_data (forest)) {
+    return 0;
+  }
+  else {
+    int32_t h = T8_DPYRAMID_LEN (level);
+    switch (ts->t8_element_shape (elements[0])) {
+    case T8_ECLASS_TET:
+      type = ((t8_dtet_t *) elements[0])->type;
+      if (type == 0 || type == 2 || type == 4) {
+        return 1;
+      }
+      else {
+        return 0;
+      }
+    case T8_ECLASS_PYRAMID:
+
+      if (!(((t8_dpyramid_t *) elements[0])->pyramid.x & h)) {
+        type = ((t8_dpyramid_t *) elements[0])->pyramid.type;
+        return type == 6 ? 1 : 0;
+      }
+      else {
+        return 0;
+      }
+    default:
+      return 1;
+    }
+  }
+}
+
+static void
+t8_basic_hybrid (int level, int endlvl, int do_vtk, t8_eclass_t eclass, int num_elements, int mesh, int balance,
```

Comment:
Please comment your code.
This function has over 130 lines and only a single comment.